source/ota.c

```diff
@@ -86,182 +95,386 @@ typedef struct OtaStateTableEntry
                                            size_t valueLength,
                                            void * pParamAdd,
                                            uint32_t * pParamSizeAdd );
 
-/* Check if all the required parameters for the job are present in the JSON. */
-
+/**
+ * @brief Check if all the required parameters for job document are extracted from the JSON.
+ *
+ * @param pModelParam Structure to store the details of keys and where to store them.
+ * @param pDocModel Details of expected parameters in the job doc.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 static DocParseErr_t verifyRequiredParamsExtracted( const JsonDocParam_t * pModelParam,
                                                     const JsonDocModel_t * pDocModel );
 
-/* Validate the version of the update received. */
-
+/**
+ * @brief Validate the version of the update received.
+ *
+ * @param pFileContext Information of file to be streamed.
+ * @return OtaErr_t
+ */
 static OtaErr_t validateUpdateVersion( const OtaFileContext_t * pFileContext );
 
-/* Check if the JSON can be parsed through a custom callback if initial parsing fails. */
-
+/**
+ * @brief Check if the JSON can be parsed through a custom callback if initial parsing fails.
+ *
+ * @param pJson JSON job document.
+ * @param messageLength Length of the job document.
+ * @param pFileContext Information of file to be streamed.
+ * @param pFinalFile File that stores all extracted params.
+ * @return OtaJobParseErr_t
+ */
 static OtaJobParseErr_t parseJobDocFromCustomCallback( const char * pJson,
                                                        uint32_t messageLength,
                                                        OtaFileContext_t * pFileContext,
                                                        OtaFileContext_t ** pFinalFile );
 
-/* Check if the incoming job document is not conflicting with current job status. */
-
+/**
+ * @brief Check if the incoming job document is not conflicting with current job status.
+ *
+ * @param pFileContext Information of file to be streamed.
+ * @param pFinalFile File that stores all extracted params.
+ * @param pUpdateJob Represents if the job is accepted.
+ * @return OtaJobParseErr_t
+ */
 static OtaJobParseErr_t verifyActiveJobStatus( OtaFileContext_t * pFileContext,
                                                OtaFileContext_t ** pFinalFile,
                                                bool * pUpdateJob );
 
-/* Check if all the file context params are valid and initialize resources for the job transfer */
-
+/**
+ * @brief Check if all the file context params are valid and initialize resources for the job transfer.
+ *
+ * @param pFileContext Information of file to be streamed.
+ * @param pFinalFile File that stores all extracted params.
+ * @param pUpdateJob Represents if the job is accepted.
+ * @return OtaJobParseErr_t
```

Comment:
Maybe somthing more descriptive?

Reply:
Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/116/commits/3e5b1d96211d212a79fd5aa4a1bd33a51add1e85 to add error descriptions